src/overdub.jl

```diff
@@ -558,8 +558,7 @@ function overdub_definition(pass, line, file)
             prehook(ctx, f, args...)
             output = execute(ctx, f, args...)
             output = isa(output, OverdubInstead) ? overdub(ctx, f, args...) : output
-            posthook(ctx, output, f, args...)
```

Comment:
Note that changing the `@inline` annotation of the enclosing `overdub` method here to `@noinline` fixes the miscompile.

However, fixing this specific miscompile causes another one to pop up in the `InferCtx` tests. It's like a perverse game whack-a-mole-form-hell.